corehq/motech/repeaters/tests/test_models_slow.py

```diff
@@ -0,0 +1,164 @@
+import random
+import string
+from collections import namedtuple
+from unittest.mock import patch
+from uuid import uuid4
+
+from django.test import TestCase
+from django.utils import timezone
+
+from requests.exceptions import ConnectionError
+
+from corehq.apps.accounting.models import SoftwarePlanEdition
+from corehq.apps.accounting.tests.utils import DomainSubscriptionMixin
+from corehq.apps.accounting.utils import clear_plan_version_cache
+from corehq.apps.domain.shortcuts import create_domain
+from corehq.apps.receiverwrapper.util import submit_form_locally
+from corehq.motech.models import ConnectionSettings
+from corehq.motech.repeaters.const import (
+    RECORD_FAILURE_STATE,
+    RECORD_SUCCESS_STATE,
+)
+from corehq.motech.repeaters.models import (
+    FormRepeater,
+    RepeaterStub,
+    send_request,
+)
+
+DOMAIN = ''.join([random.choice(string.ascii_lowercase) for __ in range(20)])
+
+
+ResponseMock = namedtuple('ResponseMock', 'status_code reason')
+
+
+class RepeaterFixtureMixin:
+
+    def setUp(self):
+        url = 'https://www.example.com/api/'
+        conn = ConnectionSettings.objects.create(domain=DOMAIN, name=url, url=url)
+        self.repeater = FormRepeater(
+            domain=DOMAIN,
+            connection_settings_id=conn.id,
+            include_app_id_param=False,
+        )
+        self.repeater.save()
+        self.repeater_stub = RepeaterStub.objects.create(
+            domain=DOMAIN,
+            repeater_id=self.repeater.get_id,
+        )
+
+    def tearDown(self):
+        self.repeater_stub.delete()
+        self.repeater.delete()
+
+
+class ServerErrorTests(RepeaterFixtureMixin, TestCase, DomainSubscriptionMixin):
```

Comment:
Are these the tests you were referring to that are slow? I'm assuming they're slow because they call `submit_form_locally()`, is that correct? If yes, I don't think there's any way to get around that if you really need to submit a form for every test.

Maybe you could submit one form in `setUpClass` and then have `tearDown` reset any form-related state that could have been changed by the test?